odoo/tools/misc.py

```diff
@@ -1363,6 +1364,28 @@ def format_duration(value):
     return '%02d:%02d' % (hours, minutes)
 
 
+def format_float(env, value, precision_rounding=None, uom=None, lang_code=False):
+    if precision_rounding is not None:
+        rounding = precision_rounding
+    elif uom is not None:
+        rounding = uom.rounding
+    else:
+        rounding = 0.01
+
+    dp = int(math.ceil(math.log10(1 / rounding))) if 0 < rounding < 1 else 0
+
+    fmt = u'%.{0}f'.format(dp)
+    lang = get_lang(env, lang_code)
+
+    formated_value = lang.format(fmt,
+                                 float_round(value, precision_rounding=rounding),
+                                 grouping=True, monetary=False)
+
+    if uom is not None and uom.name:
```

Comment:
I would maybe give an uom_name parameter and use if it defined, still to be model independent. This is a simple tool that should not depend on unit of measure.